trin-execution/src/trie_walker/filter.rs

```diff
@@ -1,120 +1,131 @@
-use alloy::primitives::{B256, U256};
+use alloy::primitives::U256;
 use rand::{thread_rng, Rng};
 
-use crate::utils::partial_nibble_path_to_right_padded_b256;
+use crate::utils::nibbles_to_right_padded_b256;
 
 #[derive(Debug, Clone)]
 pub struct Filter {
-    start_prefix: U256,
-    end_prefix: U256,
+    start: U256,
+    end: U256,
 }
 
 impl Filter {
-    pub fn new_random_filter(slice_count: u16) -> Self {
+    /// Create a new filter that includes the whole trie
+    /// Slice index must be less than slice count or it will panic
+    pub fn new(slice_index: u16, slice_count: u16) -> Self {
+        assert!(
```

Comment:
This condition would fail if `slice_count` is zero (which seems to be supported case otherwise).